### PR TITLE
Limit max Redundant Audio Encoding packet size to 1000 bytes.

### PR DESCRIPTION
**Pull request by dinmin-amzn**
**Issue #:**

**Description of changes:**
In Chromium-based browsers, writing audio frames larger than 1000 bytes will cause an error to be thrown, so we limit the max Redundant Audio Encoding packet size to 1000 bytes. See https://crbug.com/1248479.

**Testing:**
Previously, starting a call with WebAudio, Redundant Audio Encoding, and fullband music stereo quality all enabled and introducing high uplink packet loss (>20%) would cause the Redundant Audio Encoding payloads to exceed the Chromium-imposed limit and audio sending would stop working for the rest of the meeting. This issue is fixed by this PR.

*Can these be tested using a demo application? Please provide reproducible step-by-step instructions.*
1. Have two attendees join a meeting using Chrome with WebAudio, Redundant Audio Encoding, and fullband music stereo quality all enabled
2. Enable audio sending and receiving for both clients
3. Introduce >20% uplink packet loss
4. Both attendees should still be able to receive and hear audio from the other attendee

**Checklist:**

1. Have you successfully run `npm run build:release` locally?
y

5. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
n

6. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
n

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/redundantaudioencoder/RedundantAudioEncoder.ts [hunk 1]
```diff
@@ -236,12 +237,12 @@ export default class RedundantAudioEncoder {
     frame: RTCEncodedAudioFrame,
     controller: TransformStreamDefaultController
   ): void {
-    // const frameMetadata = frame.getMetadata();
```
Comment (ltrung) [inline src/redundantaudioencoder/RedundantAudioEncoder.ts:239]: Is this a miss from previous PR? Does not seem related with the issue this PR tries to fix.

Comment (pracheth) [reply src/redundantaudioencoder/RedundantAudioEncoder.ts:239]: Yeah this was missed from the previous CR

**Review by pracheth — DISMISSED** [review]

File: src/redundantaudioencoder/RedundantAudioEncoder.ts (hunk 1 above)
Comment (pracheth) [reply src/redundantaudioencoder/RedundantAudioEncoder.ts:239]: @dinmin-amzn Curious why a unit test did not catch this. Maybe we can add one?

Comment (dinmin-amzn) [reply src/redundantaudioencoder/RedundantAudioEncoder.ts:239]: Yes, let me add a unit test for this

**Review by mklingb — APPROVED** [review]

File: src/redundantaudioencoder/RedundantAudioEncoder.ts (hunk 1 above)
Comment (dinmin-amzn) [reply src/redundantaudioencoder/RedundantAudioEncoder.ts:239]: Updated the existing unit tests to better capture this case.

File: test/redundantaudioencoder/RedundantAudioEncoder.test.ts [hunk 2]
```diff
@@ -574,20 +574,51 @@ describe('RedundantAudioEncoder', () => {
...
       encoder.setRedPayloadType(redPayloadType + 1);
+      frame = createRTCEncodedAudioFrame(packetizationTime * i, redPayloadType, 0);
+      frame.data = createRedPayload([header], [importantPayload]);
       encoder['senderTransform'](frame, controller);
-      expect(frame.data.byteLength).to.equal(0);
+      expect(frame.data.byteLength).to.equal(header.byteLength + importantPayload.byteLength);
     });
 
     it('forwards frames when the primary payload cannot be determined', () => {
+      const header = createRedHeader(
```
Comment (dinmin-amzn) [inline test/redundantaudioencoder/RedundantAudioEncoder.test.ts:600]: I went ahead and made this test case more effective as well since it is similar in nature to the previous test case.

**Review by ltrung — APPROVED** [review]
